mp_rand_source.c

```diff
@@ -1,5 +1,7 @@
 #include "tommath_private.h"
-#ifdef MP_RAND_C
+/* MP_RAND_C relates to mp_rand.c, MP_PRIME_RAND_C relates tp mp_prime_rand.c
+   Both use s_mp_rand_source */
+#if defined(MP_RAND_C) || defined(MP_PRIME_RAND_C)
```

Comment:
```suggestion
#if defined(MP_RAND_SOURCE_C) || defined(S_MP_RAND_SOURCE_C)
```

This should be selected automagically via `tommath_class.h`, but if we change it as proposed this won't be the case. My suggestion is also not correct ... I guess the really correct way would be if the file looked as follows!? What do you think?

```c
#include "tommath_private.h"
#ifdef S_MP_RAND_SOURCE_C
/* LibTomMath, multiple-precision integer library -- Tom St Denis */
/* SPDX-License-Identifier: Unlicense */

mp_err(*s_mp_rand_source)(void *out, size_t size) = s_mp_rand_platform;
#endif

#ifdef MP_RAND_SOURCE_C
void mp_rand_source(mp_err(*source)(void *out, size_t size))
{
   s_mp_rand_source = (source == NULL) ? s_mp_rand_platform : source;
}
#endif
```

Or maybe this should even be two files?

Reply:
I took the liberty to push https://github.com/libtom/libtommath/tree/fix-582

Reply:
Yeah that works too.  I suggest you merge that, then close this PR